vertical-pod-autoscaler/pkg/recommender/input/history/history_provider_test.go

```diff
@@ -257,6 +262,25 @@ func TestGetMemorySamples(t *testing.T) {
 	assert.Equal(t, histories, map[model.PodID]*PodHistory{podID: podHistory})
 }
 
+func TestAuthHeader(t *testing.T) {
+	historyProvider, err := NewPrometheusHistoryProvider(getDefaultPrometheusHistoryProviderConfigForTest())
+	if err != nil {
+		t.Fatal(err)
+	}
+	authHeaderName := "Token"
+	testToken := "aaaaaaaaaaaa"
+	tokenEnvVariableName := "TEST_TOKEN"
+	os.Setenv(tokenEnvVariableName, testToken)
```

Comment:
You could use https://pkg.go.dev/testing#T.Setenv here to avoid the unset later. The defer down there is kind of out of place I think.

Reply:
done :)

Reply:
@kwiesmueller I removed the Unit test for now since it seems not to work. I also saw that there isn't any unit testing for round tripper at all and couldn't find any way on implementing it without making changes that will allow exposing the request object. Should I implement it using a different client just for sake of testing?